### PR TITLE
added a privacy modal

### PR DESCRIPTION
**Pull request by wotwot563**
## Description

Now when you try to login you are first requested consent.

## Type of change

-   [ ] Bug fix (non-breaking change which fixes an issue)
-   [x] New feature (non-breaking change which adds functionality)
-   [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
-   [ ] This change requires a documentation update

## Checklist

-   [x] My code follows the style guidelines of this project
-   [x] I have performed a self-review of my own code
-   [x] I have commented my code, particularly in hard-to-understand areas
-   [x] I have made corresponding changes to the documentation
-   [] I have added tests that prove my fix is effective or that my feature works
-   [x] I updated the changelog with an end-user readable description
-   [x] I assigned this pull request to the correct project board to update the sprint board

## Steps to Test or Reproduce

Outline the steps to test or reproduce the PR here.
These steps will be used during release testing.
click on a login provider.
then click accept

## Link to issue

Closes: #215

### CODE REVIEW

File: src/app/modules/account/login/login.component.ts [hunk 1]
```diff
@@ -17,23 +17,55 @@
...
+  private loginWithConsent(provider?: string): void {
+    const privacyConsentKeyName = 'PrivacyConsentGiven';
+    if (localStorage.getItem(privacyConsentKeyName) === 'true') {
+      this.authService.login(provider);
+    } else {
+      // set modal options
+      const modalOptions: ModalOptions = {
+        initialState: {
+          titleText: 'We value your privacy!',
+          mainText: `By clicking accept, you agree to our <a href="/policy" target="_blank">Privacy Policy</a>. You acknowledge that you have read and understand it. If you have questions or concerns you can contact us.`,
```
Comment (Brend-Smits) [inline src/app/modules/account/login/login.component.ts:57]:
```suggestion
          mainText: `By clicking accept, you agree to our <a href="/privacy" target="_blank" >Privacy Policy</a>. You acknowledge that you have read and understood it. If you have questions or concerns you can contact us.`,
```

Comment (rubenb994) [inline src/app/modules/account/login/login.component.ts:57]: Router link should be used in favor of href

File: src/app/modules/account/login/login.component.ts [hunk 2]
```diff
@@ -17,23 +17,55 @@
...
+        }
+      };
+      // Display modal
+      const modalRef = this.modalService.show(ModalAcceptGenericComponent, modalOptions);
+      // wait for the true emitter
```
Comment (rubenb994) [inline src/app/modules/account/login/login.component.ts:62]:
```suggestion
      // Wait for modal to emit true
```

File: src/app/modules/account/login/login.component.ts (hunk 1 above)
Comment (wotwot563) [reply src/app/modules/account/login/login.component.ts]: I tried this but the browser sanitizes it and removes the link with routerlink.
then i tried to sanitize it as html (that worked) but the link was not clickable.
so we settled on keeping it a href

**Pull request merged**